In [3]:
%pip install pandas

   ---------------------------------------- 0.0/9.8 MB ? eta -:--:--
   ---------------------------------------- 0.0/9.8 MB ? eta -:--:--
   ---------------------------------------- 0.1/9.8 MB 787.7 kB/s eta 0:00:13
   ---------------------------------------- 0.1/9.8 MB 655.4 kB/s eta 0:00:15
    --------------------------------------- 0.1/9.8 MB 853.3 kB/s eta 0:00:12
    --------------------------------------- 0.2/9.8 MB 807.1 kB/s eta 0:00:12
    --------------------------------------- 0.2/9.8 MB 808.4 kB/s eta 0:00:12
   - -------------------------------------- 0.3/9.8 MB 827.5 kB/s eta 0:00:12
   - -------------------------------------- 0.3/9.8 MB 842.9 kB/s eta 0:00:12
   - -------------------------------------- 0.3/9.8 MB 873.8 kB/s eta 0:00:11
   - -------------------------------------- 0.4/9.8 MB 882.8 kB/s eta 0:00:11
   - -------------------------------------- 0.4/9.8 MB 873.3 kB/s eta 0:00:11
   - -------------------------------------- 0.5/9.8 MB 893.0 kB/s eta 0:00:11
   -


[notice] A new release of pip is available: 24.0 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [8]:
import os
print(os.getcwd())

c:\Users\gandi\Downloads\archive


In [11]:
df = pd.read_csv("Healthcare Data Analysis for readmission.csv")

In [12]:
print("Rows and Columns:", df.shape)

Rows and Columns: (10000, 26)


In [13]:
print(df.columns.tolist())

['hospital_name', 'Admission_date', 'hospital_id', 'hospital_beds_available', 'occupied_beds', 'hospital_ward', 'patient_id', 'patient_gender', 'patient_age', 'patient_race', 'patient_sat_score', 'patient_first_initial', 'patient_last_name', 'patient_waittime', 'department_referral', 'time_slot', 'doctor_id', 'doctor_name', 'doctor_specialty', 'patient_assigned_doctor', 'patient_checkin_date', 'patient_checkout_date', 'patient_disease', 'patient_length_of_stay', 'discharge_status', 'readmission']


In [14]:
display(df.head())


,hospital_name,Admission_date,hospital_id,hospital_beds_available,occupied_beds,hospital_ward,patient_id,patient_gender,patient_age,patient_race,...,doctor_id,doctor_name,doctor_specialty,patient_assigned_doctor,patient_checkin_date,patient_checkout_date,patient_disease,patient_length_of_stay,discharge_status,readmission
0,The Johns Hopkins Hospital,30-01-2024,3946,260,90,Maternity,1421,Female,48,Asian,...,1725,Justin Morris,Rheumatology,False,14-07-2024,19-07-2024,Anxiety Disorders,6,Deceased,1
1,The Johns Hopkins Hospital,20-03-2022,7147,250,90,ICU,4922,Male,40,Black,...,7510,Latoya Moss,Pulmonology or Allergy and Immunology,False,19-07-2024,07/07/2024,Urinary Tract Infection (UTI),2,Deceased,1
2,The Johns Hopkins Hospital,13-07-2021,4174,350,220,Pediatrics,9804,Female,74,Black,...,7137,Michael Allen,Neurology,False,20-07-2024,29-06-2024,Hypertension (High Blood Pressure),17,Recovered,0
3,The Johns Hopkins Hospital,26-01-2021,2466,190,350,Pediatrics,5622,Male,82,Hispanic,...,8801,Rebecca Mccullough,Cardiology,False,25-06-2024,16-07-2024,Arthritis,28,Recovered,0
4,The Johns Hopkins Hospital,06/02/2024,3014,220,390,Surgery,3314,Male,62,Hispanic,...,3588,John Giles,Rheumatology,False,21-07-2024,19-07-2024,Cancer,16,Transferred,1


In [15]:
print(df.dtypes)

hospital_name                str
Admission_date               str
hospital_id                int64
hospital_beds_available    int64
occupied_beds              int64
hospital_ward                str
patient_id                 int64
patient_gender               str
patient_age                int64
patient_race                 str
patient_sat_score          int64
patient_first_initial        str
patient_last_name            str
patient_waittime           int64
department_referral          str
time_slot                    str
doctor_id                  int64
doctor_name                  str
doctor_specialty             str
patient_assigned_doctor     bool
patient_checkin_date         str
patient_checkout_date        str
patient_disease              str
patient_length_of_stay     int64
discharge_status             str
readmission                int64
dtype: object


In [16]:
print(df.isnull().sum())

hospital_name              0
Admission_date             0
hospital_id                0
hospital_beds_available    0
occupied_beds              0
hospital_ward              0
patient_id                 0
patient_gender             0
patient_age                0
patient_race               0
patient_sat_score          0
patient_first_initial      0
patient_last_name          0
patient_waittime           0
department_referral        0
time_slot                  0
doctor_id                  0
doctor_name                0
doctor_specialty           0
patient_assigned_doctor    0
patient_checkin_date       0
patient_checkout_date      0
patient_disease            0
patient_length_of_stay     0
discharge_status           0
readmission                0
dtype: int64


In [17]:
print("Duplicate Rows:", df.duplicated().sum())

Duplicate Rows: 0


In [18]:
print(df.describe())

        hospital_id  hospital_beds_available  occupied_beds    patient_id  \
count  10000.000000             10000.000000   10000.000000  10000.000000   
mean    5515.966600               299.111000     199.085000   5498.738700   
std     2605.495574               117.973859     118.299418   2615.168591   
min     1000.000000               100.000000       0.000000   1000.000000   
25%     3255.750000               200.000000     100.000000   3232.500000   
50%     5516.500000               300.000000     200.000000   5478.500000   
75%     7790.250000               400.000000     300.000000   7779.500000   
max     9998.000000               500.000000     400.000000   9999.000000   

        patient_age  patient_sat_score  patient_waittime     doctor_id  \
count  10000.000000       10000.000000      10000.000000  10000.000000   
mean      47.283000         847.802000         67.824000   5476.229000   
std       27.719692         432.952877         34.519765   2626.286129   
min       

In [19]:
print(df["readmission"].value_counts())

readmission
0    5858
1    4142
Name: count, dtype: int64


In [20]:
print(df["hospital_ward"].value_counts())

hospital_ward
Pediatrics    2067
ER            2037
ICU           1971
Maternity     1967
Surgery       1958
Name: count, dtype: int64


In [21]:
print(df["department_referral"].value_counts())

department_referral
Neurology                                  1790
Gastroenterology                           1094
Psychiatry                                  730
Rheumatology                                720
Infectious Disease or Internal Medicine     700
Pulmonology or Allergy and Immunology       380
Cardiology                                  374
Hepatology or Gastroenterology              371
Hepatology                                  367
Infectious Disease                          362
Allergy and Immunology                      359
Pulmonology                                 354
Urology or Nephrology                       351
Oncology                                    349
Endocrinology                               347
Cardiology or Internal Medicine             346
Dermatology                                 339
Pulmonology or Infectious Disease           336
Nephrology                                  331
Name: count, dtype: int64


In [22]:
print(df["doctor_specialty"].value_counts())

doctor_specialty
Neurology                                  1792
Gastroenterology                           1026
Rheumatology                                767
Psychiatry                                  724
Infectious Disease or Internal Medicine     703
Endocrinology                               373
Nephrology                                  372
Pulmonology or Infectious Disease           370
Cardiology or Internal Medicine             369
Hepatology or Gastroenterology              368
Pulmonology or Allergy and Immunology       359
Cardiology                                  356
Oncology                                    356
Dermatology                                 354
Allergy and Immunology                      353
Urology or Nephrology                       350
Pulmonology                                 341
Hepatology                                  338
Infectious Disease                          329
Name: count, dtype: int64


In [23]:
print(df["Admission_date"].head(10))

0    30-01-2024
1    20-03-2022
2    13-07-2021
3    26-01-2021
4    06/02/2024
5    29-04-2022
6    25-07-2020
7    18-05-2022
8    20-07-2022
9    29-05-2022
Name: Admission_date, dtype: str


In [24]:
print(df["patient_checkin_date"].head(10))

0    14-07-2024
1    19-07-2024
2    20-07-2024
3    25-06-2024
4    21-07-2024
5    03/07/2024
6    21-07-2024
7    19-07-2024
8    09/07/2024
9    14-07-2024
Name: patient_checkin_date, dtype: str


In [25]:
print(df["patient_checkout_date"].head(10))

0    19-07-2024
1    07/07/2024
2    29-06-2024
3    16-07-2024
4    19-07-2024
5    25-06-2024
6    16-07-2024
7    14-07-2024
8    18-07-2024
9    12/07/2024
Name: patient_checkout_date, dtype: str


In [26]:
print(df["patient_gender"].value_counts())

patient_gender
Male      5012
Female    4988
Name: count, dtype: int64


In [27]:
df_clean = df.copy()

In [29]:
date_cols = [
    "Admission_date",
    "patient_checkin_date",
    "patient_checkout_date"
]

for col in date_cols:
    df_clean[col] = pd.to_datetime(
        df_clean[col],
        format="mixed",
        dayfirst=True
    )

In [30]:
df_clean.dtypes

hospital_name                         str
Admission_date             datetime64[us]
hospital_id                         int64
hospital_beds_available             int64
occupied_beds                       int64
hospital_ward                         str
patient_id                          int64
patient_gender                        str
patient_age                         int64
patient_race                          str
patient_sat_score                   int64
patient_first_initial                 str
patient_last_name                     str
patient_waittime                    int64
department_referral                   str
time_slot                             str
doctor_id                           int64
doctor_name                           str
doctor_specialty                      str
patient_assigned_doctor              bool
patient_checkin_date       datetime64[us]
patient_checkout_date      datetime64[us]
patient_disease                       str
patient_length_of_stay            

In [31]:
df_clean.isnull().sum()


hospital_name              0
Admission_date             0
hospital_id                0
hospital_beds_available    0
occupied_beds              0
hospital_ward              0
patient_id                 0
patient_gender             0
patient_age                0
patient_race               0
patient_sat_score          0
patient_first_initial      0
patient_last_name          0
patient_waittime           0
department_referral        0
time_slot                  0
doctor_id                  0
doctor_name                0
doctor_specialty           0
patient_assigned_doctor    0
patient_checkin_date       0
patient_checkout_date      0
patient_disease            0
patient_length_of_stay     0
discharge_status           0
readmission                0
dtype: int64

In [32]:
df_clean.duplicated().sum()

np.int64(0)

In [33]:
df_clean["hospital_ward"].value_counts()

hospital_ward
Pediatrics    2067
ER            2037
ICU           1971
Maternity     1967
Surgery       1958
Name: count, dtype: int64

In [34]:
df_clean["hospital_ward"] = df_clean["hospital_ward"].replace({
    "ER": "Emergency Department"
})

In [35]:
df_clean["hospital_ward"].value_counts()

hospital_ward
Pediatrics              2067
Emergency Department    2037
ICU                     1971
Maternity               1967
Surgery                 1958
Name: count, dtype: int64

In [36]:
df_clean["patient_id"].nunique()

6022

In [37]:
df_clean["department_referral"].value_counts()

department_referral
Neurology                                  1790
Gastroenterology                           1094
Psychiatry                                  730
Rheumatology                                720
Infectious Disease or Internal Medicine     700
Pulmonology or Allergy and Immunology       380
Cardiology                                  374
Hepatology or Gastroenterology              371
Hepatology                                  367
Infectious Disease                          362
Allergy and Immunology                      359
Pulmonology                                 354
Urology or Nephrology                       351
Oncology                                    349
Endocrinology                               347
Cardiology or Internal Medicine             346
Dermatology                                 339
Pulmonology or Infectious Disease           336
Nephrology                                  331
Name: count, dtype: int64

In [38]:
df_clean["department_referral"].unique()

<StringArray>
[                  'Urology or Nephrology',
                               'Neurology',
                        'Gastroenterology',
                             'Dermatology',
 'Infectious Disease or Internal Medicine',
                  'Allergy and Immunology',
                              'Psychiatry',
                      'Infectious Disease',
                              'Nephrology',
                              'Cardiology',
                            'Rheumatology',
         'Cardiology or Internal Medicine',
                                'Oncology',
   'Pulmonology or Allergy and Immunology',
          'Hepatology or Gastroenterology',
                           'Endocrinology',
                              'Hepatology',
                             'Pulmonology',
       'Pulmonology or Infectious Disease']
Length: 19, dtype: str

In [39]:
df_clean["doctor_specialty"].value_counts()

doctor_specialty
Neurology                                  1792
Gastroenterology                           1026
Rheumatology                                767
Psychiatry                                  724
Infectious Disease or Internal Medicine     703
Endocrinology                               373
Nephrology                                  372
Pulmonology or Infectious Disease           370
Cardiology or Internal Medicine             369
Hepatology or Gastroenterology              368
Pulmonology or Allergy and Immunology       359
Cardiology                                  356
Oncology                                    356
Dermatology                                 354
Allergy and Immunology                      353
Urology or Nephrology                       350
Pulmonology                                 341
Hepatology                                  338
Infectious Disease                          329
Name: count, dtype: int64

In [40]:
df_clean["patient_age"].min(), df_clean["patient_age"].max()

(np.int64(0), np.int64(95))

In [41]:
(df_clean["patient_age"] == 0).sum()

np.int64(94)

In [42]:
df_clean["patient_length_of_stay"].min(), df_clean["patient_length_of_stay"].max()

(np.int64(1), np.int64(30))

In [43]:
(df_clean["occupied_beds"] > df_clean["hospital_beds_available"]).sum()

np.int64(2777)

In [44]:
df_clean.loc[
    df_clean["occupied_beds"] > df_clean["hospital_beds_available"],
    ["hospital_beds_available", "occupied_beds"]
].head(10)

,hospital_beds_available,occupied_beds
3,190,350
4,220,390
7,350,370
11,240,350
25,240,310
37,290,380
42,110,140
50,130,280
54,220,340
57,110,330


In [45]:
(df_clean["hospital_beds_available"] < 0).sum()

np.int64(0)

In [46]:
(df_clean["occupied_beds"] < 0).sum()

np.int64(0)

In [47]:
df_clean["readmission"].value_counts()

readmission
0    5858
1    4142
Name: count, dtype: int64

In [48]:
df_clean["patient_waittime"].min(), df_clean["patient_waittime"].max()

(np.int64(15), np.int64(120))

In [49]:
df_clean["patient_sat_score"].min(), df_clean["patient_sat_score"].max()

(np.int64(100), np.int64(1600))

In [50]:
df_clean["hospital_id"].nunique()

6074

In [51]:
df_clean["doctor_id"].nunique()

6019

In [52]:
(df_clean["doctor_id"] < 0).sum()

np.int64(0)

In [53]:
(df_clean["hospital_id"] < 0).sum()

np.int64(0)

In [54]:
df_clean["patient_gender"].value_counts()

patient_gender
Male      5012
Female    4988
Name: count, dtype: int64

In [55]:
df_clean["discharge_status"].value_counts()

discharge_status
Recovered      3384
Deceased       3319
Transferred    3297
Name: count, dtype: int64

In [57]:
df_clean["time_slot"].value_counts()

time_slot
05:56:00 AM    18
07:33:00 AM    17
12:45:00 PM    17
09:33:00 PM    16
09:39:00 AM    16
               ..
07:58:00 AM     1
08:14:00 PM     1
06:17:00 AM     1
10:34:00 AM     1
11:37:00 PM     1
Name: count, Length: 1439, dtype: int64

In [58]:
df_clean["patient_assigned_doctor"].value_counts()

patient_assigned_doctor
False    5039
True     4961
Name: count, dtype: int64

In [59]:
(df_clean["patient_checkout_date"] < df_clean["patient_checkin_date"]).sum()

np.int64(4750)

In [60]:
df_clean.loc[
    df_clean["patient_checkout_date"] < df_clean["patient_checkin_date"],
    ["patient_checkin_date", "patient_checkout_date", "patient_length_of_stay"]
].head(10)

,patient_checkin_date,patient_checkout_date,patient_length_of_stay
1,2024-07-19,2024-07-07,2
2,2024-07-20,2024-06-29,17
4,2024-07-21,2024-07-19,16
5,2024-07-03,2024-06-25,18
6,2024-07-21,2024-07-16,29
7,2024-07-19,2024-07-14,3
9,2024-07-14,2024-07-12,2
12,2024-07-12,2024-06-27,14
13,2024-07-15,2024-06-28,5
18,2024-07-01,2024-06-30,23


In [61]:
df_clean["Admission_date"].min(), df_clean["Admission_date"].max()

(Timestamp('2020-07-24 00:00:00'), Timestamp('2024-07-22 00:00:00'))

In [62]:
df_clean.shape

(10000, 26)

In [63]:
df_clean.head()

,hospital_name,Admission_date,hospital_id,hospital_beds_available,occupied_beds,hospital_ward,patient_id,patient_gender,patient_age,patient_race,...,doctor_id,doctor_name,doctor_specialty,patient_assigned_doctor,patient_checkin_date,patient_checkout_date,patient_disease,patient_length_of_stay,discharge_status,readmission
0,The Johns Hopkins Hospital,2024-01-30,3946,260,90,Maternity,1421,Female,48,Asian,...,1725,Justin Morris,Rheumatology,False,2024-07-14,2024-07-19,Anxiety Disorders,6,Deceased,1
1,The Johns Hopkins Hospital,2022-03-20,7147,250,90,ICU,4922,Male,40,Black,...,7510,Latoya Moss,Pulmonology or Allergy and Immunology,False,2024-07-19,2024-07-07,Urinary Tract Infection (UTI),2,Deceased,1
2,The Johns Hopkins Hospital,2021-07-13,4174,350,220,Pediatrics,9804,Female,74,Black,...,7137,Michael Allen,Neurology,False,2024-07-20,2024-06-29,Hypertension (High Blood Pressure),17,Recovered,0
3,The Johns Hopkins Hospital,2021-01-26,2466,190,350,Pediatrics,5622,Male,82,Hispanic,...,8801,Rebecca Mccullough,Cardiology,False,2024-06-25,2024-07-16,Arthritis,28,Recovered,0
4,The Johns Hopkins Hospital,2024-02-06,3014,220,390,Surgery,3314,Male,62,Hispanic,...,3588,John Giles,Rheumatology,False,2024-07-21,2024-07-19,Cancer,16,Transferred,1


In [64]:
df_clean.to_csv("readmission_cleaned.csv", index=False)

In [65]:
import os
os.path.exists("readmission_cleaned.csv")

True

In [66]:
os.getcwd()

'c:\\Users\\gandi\\Downloads\\archive'

In [67]:
import os

os.makedirs("data/processed", exist_ok=True)

In [68]:
df_clean.to_csv("data/processed/readmission_cleaned.csv", index=False)

In [69]:
os.path.exists("data/processed/readmission_cleaned.csv")

True

In [70]:
df_clean.info()

<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 26 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   hospital_name            10000 non-null  str           
 1   Admission_date           10000 non-null  datetime64[us]
 2   hospital_id              10000 non-null  int64         
 3   hospital_beds_available  10000 non-null  int64         
 4   occupied_beds            10000 non-null  int64         
 5   hospital_ward            10000 non-null  str           
 6   patient_id               10000 non-null  int64         
 7   patient_gender           10000 non-null  str           
 8   patient_age              10000 non-null  int64         
 9   patient_race             10000 non-null  str           
 10  patient_sat_score        10000 non-null  int64         
 11  patient_first_initial    10000 non-null  str           
 12  patient_last_name        10000 non-null  str

## Readmission Dataset – Profiling, Cleaning & Validation Summary

### Dataset Overview
- Dataset contains 10,000 rows and 26 columns.
- No missing values were found.
- No completely duplicate records were found.
- Admission_date, patient_checkin_date, and patient_checkout_date were converted to datetime format.

### Data Cleaning
- Hospital ward values were standardized by changing "ER" to "Emergency Department".
- Patient age was checked; values range from 0 to 95.
- Patient length of stay ranges from 1 to 30 days.
- Readmission contains valid binary values: 0 and 1.
- Negative hospital bed, occupied bed, hospital ID, and doctor ID values were not found.
- Patient gender and discharge status values were checked.

### Data Consistency Issues Identified

1. **Hospital Name vs Hospital ID**
   - Only 1 unique hospital name was found.
   - The hospital name "The Johns Hopkins Hospital" is associated with 6,074 different hospital IDs.
   - Original hospital IDs were preserved because the correct mapping could not be determined from the source data.

2. **Checkout Date Earlier Than Check-in Date**
   - 4,750 records have patient_checkout_date earlier than patient_checkin_date.
   - Original date values were preserved because the correct dates could not be determined reliably.

3. **Length of Stay Mismatch**
   - 9,822 records have a mismatch between the calculated date difference and patient_length_of_stay.
   - LOS values were not modified because the check-in and checkout dates themselves contain inconsistencies.

4. **Admission Date Later Than Check-in Date**
   - 87 records have Admission_date later than patient_checkin_date.
   - Original values were preserved because the correct date relationship could not be determined from the available source information.

5. **Doctor ID Mapped to Multiple Doctor Names**
   - 2,755 doctor IDs are associated with more than one doctor name.
   - Some IDs are also associated with different doctor specialties.
   - Example: Doctor ID 1002 is associated with three different doctor names and specialties.
   - Original doctor IDs and names were preserved because the correct mapping could not be determined reliably.

### Validation Decision
- The identified consistency issues were verified using Python checks.
- No values were arbitrarily changed or deleted.
- The original source values were preserved where reliable correction evidence was unavailable.
- The identified issues are documented for further validation with the source data/team.

### Output
- Final cleaned dataset contains 10,000 rows and 26 columns.
- Cleaned dataset saved as: `data/processed/readmission_cleaned.csv`

In [71]:
df_clean[["hospital_name", "hospital_id"]].drop_duplicates().head(20)

,hospital_name,hospital_id
0,The Johns Hopkins Hospital,3946
1,The Johns Hopkins Hospital,7147
2,The Johns Hopkins Hospital,4174
3,The Johns Hopkins Hospital,2466
4,The Johns Hopkins Hospital,3014
5,The Johns Hopkins Hospital,9743
6,The Johns Hopkins Hospital,8842
7,The Johns Hopkins Hospital,4385
8,The Johns Hopkins Hospital,8087
9,The Johns Hopkins Hospital,1132


In [72]:
df_clean["hospital_name"].nunique()


1

In [73]:
df_clean["hospital_id"].nunique()

6074

In [74]:
df_clean.groupby("hospital_name")["hospital_id"].nunique()

hospital_name
The Johns Hopkins Hospital    6074
Name: hospital_id, dtype: int64

In [75]:
(df_clean["patient_checkout_date"] < df_clean["patient_checkin_date"]).sum()

np.int64(4750)

In [76]:
df_clean.loc[
    df_clean["patient_checkout_date"] < df_clean["patient_checkin_date"],
    ["patient_id", "patient_checkin_date", "patient_checkout_date", "patient_length_of_stay"]
].head(10)

,patient_id,patient_checkin_date,patient_checkout_date,patient_length_of_stay
1,4922,2024-07-19,2024-07-07,2
2,9804,2024-07-20,2024-06-29,17
4,3314,2024-07-21,2024-07-19,16
5,3222,2024-07-03,2024-06-25,18
6,8824,2024-07-21,2024-07-16,29
7,4772,2024-07-19,2024-07-14,3
9,7767,2024-07-14,2024-07-12,2
12,7767,2024-07-12,2024-06-27,14
13,5747,2024-07-15,2024-06-28,5
18,5135,2024-07-01,2024-06-30,23


In [77]:
(df_clean["patient_checkout_date"] - df_clean["patient_checkin_date"]).dt.days.head(10)

0     5
1   -12
2   -21
3    21
4    -2
5    -8
6    -5
7    -5
8     9
9    -2
dtype: int64

In [78]:
date_los = (
    df_clean["patient_checkout_date"] -
    df_clean["patient_checkin_date"]
).dt.days

(date_los != df_clean["patient_length_of_stay"]).sum()

np.int64(9822)

In [79]:
(df_clean["Admission_date"] > df_clean["patient_checkin_date"]).sum()

np.int64(87)

In [2]:
import pandas as pd

df_clean = pd.read_csv(
    "data/processed/readmission_cleaned.csv"
)

In [3]:
doctor_name_count = df_clean.groupby("doctor_id")["doctor_name"].nunique()

(doctor_name_count > 1).sum()

np.int64(2755)

In [4]:
doctor_name_count[doctor_name_count > 1].head(10)

doctor_id
1002    3
1010    2
1011    3
1012    2
1013    3
1023    2
1025    2
1027    4
1028    2
1031    2
Name: doctor_name, dtype: int64

In [5]:
df_clean.loc[
    df_clean["doctor_id"] == 1002,
    ["doctor_id", "doctor_name", "doctor_specialty"]
].drop_duplicates()

,doctor_id,doctor_name,doctor_specialty
7478,1002,Blake Ryan,Psychiatry
8671,1002,Patricia Lee,Gastroenterology
9203,1002,Kenneth Bell,Nephrology
